### PR TITLE
[RFC] Command Line Interface (CLI)

### PR DESCRIPTION
**Pull request by chrisdmacrae**

### CODE REVIEW

**Review by jssee — COMMENTED** [review]
I think the CLI should also include a separate command that handles bundling `node_modules` into a local `web_modules` folder. basically what snowpack does. having it as a separate command allows us to use it as a `postinstall` hook as well, which is nice.

**Comment by jssee** [comment]
> I think the CLI should also include a separate command that handles bundling `node_modules` into a local `web_modules` folder. basically what snowpack does. having it as a separate command allows us to use it as a `postinstall` hook as well, which is nice.

right now I have it defined as `spike pack` in the [cli skeleton](https://github.com/spike-js/spike/blob/6d0746a7275f839d2848ea5b76d668d24a90e182/src/prog.ts#L16)

**Comment by chrisdmacrae** [comment]
I like this. 👍

File: architecture/cli.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ The CLI for Spike is a simple CLI interface over the core NodeJS API.
 The CLI will need to facilitate the following verbs/commands:
 
 - Creating a site (`init`|`initialize`)
+- Pack a sites `web_modules`
```
Comment (jssee) [inline architecture/cli.md]: should we define the command here? 

```suggestion
- Pack a sites `web_modules` (`pack` )
```

**Pull request merged**